### PR TITLE
Change license and update leftovers

### PR DESCRIPTION
**Pull request by realloc**
- Change license to Apache 2.0 to match SDK
- Simplify Makefile
- Update contribution guide

### CODE REVIEW

**Review by anatoly-bogatyrev — APPROVED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+
+Description
+
+<Macros>
+
+<Sign-Off>
+```
+
+```
+$ git commit -sam '[#123] Add some feature'
```
Comment (fyrchik) [inline CONTRIBUTING.md:81]: This repo historically does not add issue numbers. Ironically, this PR too. Do we want to start?
I would stay with the old approach given the amount of changes we have here.

File: LICENSE [hunk 2]
```diff
@@ -1,674 +1,201 @@
...
-the library.  If this is what you want to do, use the GNU Lesser General
-Public License instead of this License.  But first, please read
-<https://www.gnu.org/licenses/why-not-lgpl.html>.
+                                 Apache License
+                           Version 2.0, January 2004
+                        http://www.apache.org/licenses/
+
+   TERMS AND CONDITIONS FOR USE, REPRODUCTION, AND DISTRIBUTION
+
+   1. Definitions.
```
Comment (fyrchik) [inline LICENSE:7]: > Written approvals from all original authors were received

RLY? :smile:

File: demo.sh [hunk 3]
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+#set -x
+
+BLOCK_SIZE=${1:-100M} # 100Mb by default
+TMPDIR="${TMPDI:-$(mktemp -d)}"
```
Comment (fyrchik) [inline demo.sh]: `TMPDI`?

File: demo.sh [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+OUT="${OUT:-"${TMPDIR}/bighash"}"
+
+echo "Preparing big file at ${OUT}..."
+dd if=/dev/urandom of="$OUT" bs="$BLOCK_SIZE" count=1
+
+echo "Make 4 smaller parts from ${OUT}..."
+split -dn 4 ${OUT} ${TMPDIR}/
+
+echo -n "Big file hash: "
+TZALL=`./bin/tzsum -impl avx2 -name ${OUT} | awk '{print $1}'`
```
Comment (fyrchik) [inline demo.sh]: Why backtick and not `$()`?
`shellcheck` usually complains.

File: CONTRIBUTING.md (hunk 1 above)
Comment (realloc) [reply CONTRIBUTING.md:81]: I propose to use a common style across all repositories here.

File: LICENSE (hunk 2 above)
Comment (realloc) [reply LICENSE:7]: Of course! =)

File: demo.sh (hunk 4 above)
Comment (realloc) [reply demo.sh]: Agree. A blast from the past =)

File: CONTRIBUTING.md (hunk 1 above)
Comment (realloc) [reply CONTRIBUTING.md:81]: And I hope we could have more changes coming.

Comment (fyrchik) [reply CONTRIBUTING.md:81]: Then again, what about this PR?

Comment (realloc) [reply CONTRIBUTING.md:81]: Added proper references in commit summary. I promise to split similar bulk changes in to several logically smaller PRs next time =)

**Review by fyrchik — APPROVED** [review]

**Pull request merged**